src/ORM/DataList.php

```diff
@@ -501,6 +501,13 @@ public function filterByCallback($callback)
      * Unlike getRelationName, this is immutable and will fallback to the quoted field
      * name if not a relation.
      *
+     * Example use (simple WHERE condition on data sitting in a related table):
+     *
+     * <code>
+     *  $columnName = null;
+     *  $list = Page::get()->applyRelation('TaxonomyTerms.ID', $columnName)->where([$columnName => <some_value>]);
+     * </code>
+     *
```

Comment:
Oh no, I wrote this method didn't I?

I should have done a better job of documenting this sorry.